### PR TITLE
Deprecation warning: arrangement_version <= 5

### PR DESCRIPTION
**Pull request by MartinBasti**
Jira: OSBS-6345

Signed-off-by: Martin Bašti <mbasti@redhat.com>

### CODE REVIEW

File: osbs/api.py [hunk 1]
```diff
@@ -80,6 +80,20 @@ def catch_exceptions(*args, **kwargs):
...
+    """Validate if the arrangement_version is supported
+
+    Shows a warning when version is deprecated
+
+    :param int arrangement_version: version to be validated
+    :raises ValueError: when version is not supported
+    """
+    if arrangement_version <= 5:
+        # TODO: raise as ValueError in release 0.54+
+        logger.warning("arrangement_version <= 5 is deprecated and will be removed"
```
Comment (MartinBasti) [inline osbs/api.py]: Do we want issue `warnings.warn` instead of `logger.warning` given that this is executed in koji/builders?

**Comment by athos-ribeiro** [comment]
The failing tests were caused by a non-rebased merge regarding the `delete_from_registry` deactivation. I am fixing the broken tests and will request a rebuild here ASAP :)

File: tests/test_api.py [hunk 2]
```diff
@@ -96,6 +96,22 @@ class MockDfParser(object):
     baseimage = 'fedora23/python'
 
 
+@pytest.mark.parametrize('version,warning,exception', (
+    (5, "arrangement_version <= 5 is deprecated and will be removed in release 0.54", None),
```
[the remark this answers is not in the document]
Comment (MartinBasti) [reply tests/test_api.py]: It is currently only warning, no exception yet.

File: tests/test_api.py [hunk 3]
```diff
@@ -96,6 +96,22 @@ class MockDfParser(object):
...
+    (6, None, None),
+))
+def test_validate_arrangement_version(recwarn, version, warning, exception):
+    """Test deprecation mechanism of arrangement version"""
+    if exception is not None:
```
[the remark this answers is not in the document]
Comment (MartinBasti) [reply tests/test_api.py]: Updated

**Comment by MartinBasti** [comment]
Release notes:

> ## API changes
> - `arrangement_version` <= 5 is deprecated and support will be removed in version 0.54

**Pull request merged**